### ISSUE TITLE
Guest customers using registerUserOnOrderComplete results in orders with no customer id

### ISSUE THREAD

**Issue by Mathias-Syversen** [craftcms/commerce#1483]
Craft CMS 3.4.20
Craft Commerce 3.1.5

I'm currently having an issue where many orders are being generated without a customer id.
This causes a custom order event to fail, because it cant save an order without any customer id.

From what I can see this happened after I updated to commerce 3.1.4.
I got some help from Luke to find orders with no customer id, and reference the order email with existing users (who had no customer record). We fixed these, but the problem continues.

I read on discord that someone else had a problem where customers where deleted when using "registerUserOnOrderComplete" during checkout, but I cant seem to find this message. 
Maybe this can be related?

The problem occurs when a guest uses an email related to an existing user during checkout, I have found one example of this happening two times to the same "email" making me think the customer record gets deleted somehow?

I am currently rewriting the checkout to force users to log in if the order email is an existing user.

This has also caused another problem, where I cannot save **any** order in the CP see this issue #1484

**Comment by Mathias-Syversen** [comment]
Update:

I have now tried these combos:
User logged in
Guest with email of user, no registerUserOnOrderComplete
Guest with email of user, with registerUserOnOrderComplete
Guest with "new" email, no registerUserOnOrderComplete
Guest with "new" email, with registerUserOnOrderComplete

Only the last attempt, guest with "new" email with registerUserOnOrderComplete resulted in an order without customer id, i did this twice to confirm.

**Comment by lukeholder** [comment]
@Mathias-Syversen Thanks for the reporting. Working on it now.

**Comment by lukeholder** [comment]
I believe we fixed this is in 18c1671781bd767e18f852b3f618bb088b33abbf

To get the fix early, change your `craftcms/commerce` requirement in composer.json to:

```json
"require": {
  "craftcms/commerce": "dev-develop#ff96012d948da252b5388d1462fce6b298a5974d as 3.1.5",
  "...": "..."
}
```

Then run `composer update`.

We will release a fix to the order data in the next proper release.
